Error publishing Net MAUI App VS Windows paired with Mac

---
**Issue moved from dotnet/maui#21701**
- Please respond to @rspulini.

---

_From @rspulini on Monday, April 8, 2024 12:15:18 PM_

### Description

I'm using Visual Studio version 17.9.5

Net Maui 8.0 iOS, on Mac XCode version 15.3

The application worked perfectly on the Emulator pair to Mac.

I configured the project for builder according to Microsoft documentation

In the linker I configured it to: Link SDK assemblies only

dotnet publish -f net8.0-ios -c Release -p:ArchiveOnBuild=true -p:RuntimeIdentifier=ios-arm64 -p:CodesignKey="my data" -p:CodesignProvision="my data" -p:ServerAddress=my data -p:ServerUser=my data -p:ServerPassword=my data -p:TcpPort=58181 -p:_DotNetRootRemoteDirectory=/Users/Coala/Library/Caches/Xamarin/XMA/SDKs/dotnet/

**Note: If you set the linker to Don't link, it works. However, when submitting to the Appe Store, the App is rejected due to lack of keys in the info.plist**

However, when publishing the App, this error appears:

Could you please help me?
The xcrun tool execution has completed (exit code = 1).
Undefined symbols for architecture arm64:
"_SSLCopyALPNProtocols", referenced from:
_AppleCryptoNative_SslGetAlpnSelected in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLCopyDistinguishedNames", referenced from:
_AppleCryptoNative_SslCopyCADistinguishedNames in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLCopyPeerTrust", referenced from:
_AppleCryptoNative_SslCopyCertChain in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
_AppleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLCreateContext", referenced from:
_AppleCryptoNative_SslCreateContext in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLGetNegotiatedCipher", referenced from:
_AppleCryptoNative_SslGetCipherSuite in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLGetNegotiatedProtocolVersion", referenced from:
_AppleCryptoNative_SslGetProtocolVersion in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLGetPeerDomainNameLength", referenced from:
_AppleCryptoNative_SslSetTargetName in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLGetSessionState", referenced from:
_AppleCryptoNative_SslRead in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLHandshake", referenced from:
_AppleCryptoNative_SslHandshake in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLRead", referenced from:
_AppleCryptoNative_SslRead in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLSetALPNProtocols", referenced from:
_AppleCryptoNative_SSLSetALPNProtocols in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
_AppleCryptoNative_SSLSetALPNProtocol in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
(maybe you meant: _p_122_plt_System_Net_Security_Interop_AppleCrypto_SSLSetALPNProtocols_System_Net_SafeSslHandl
e_Microsoft_Win32_SafeHandles_SafeCreateHandle_int__llvm, _p_964_plt_System_Net_Security_Interop_AppleCrypto__SSLSetA
LPNProtocolsg____PInvoke_29_0_intptr_intptr_int__llvm , _AppleCryptoNative_SSLSetALPNProtocols )
"_SSLSetCertificate", referenced from:
_AppleCryptoNative_SslSetCertificate in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLSetClientSideAuthenticate", referenced from:
_AppleCryptoNative_SslSetAcceptClientCert in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLSetConnection", referenced from:
_AppleCryptoNative_SslSetConnection in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLSetEnabledCiphers", referenced from:
_AppleCryptoNative_SslSetEnabledCipherSuites in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLSetIOFuncs", referenced from:
_AppleCryptoNative_SslSetIoCallbacks in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLSetPeerDomainName", referenced from:
_AppleCryptoNative_SslSetTargetName in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLSetProtocolVersionMax", referenced from:
_AppleCryptoNative_SslSetMaxProtocolVersion in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLSetProtocolVersionMin", referenced from:
_AppleCryptoNative_SslSetMinProtocolVersion in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLSetSessionOption", referenced from:
_AppleCryptoNative_SslSetBreakOnServerAuth in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
_AppleCryptoNative_SslSetBreakOnCertRequested in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
_AppleCryptoNative_SslSetBreakOnClientAuth in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
_AppleCryptoNative_SslSetBreakOnClientHello in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SSLWrite", referenced from:
_AppleCryptoNative_SslWrite in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SecCertificateCopyData", referenced from:
_AppleCryptoNative_X509GetRawData in libSystem.Security.Cryptography.Native.Apple.a(pal_x509.c.o)
"_SecCertificateCopyKey", referenced from:
_AppleCryptoNative_X509GetPublicKey in libSystem.Security.Cryptography.Native.Apple.a(pal_x509.c.o)
"_SecCertificateCopyPublicKey", referenced from:
_AppleCryptoNative_X509GetPublicKey in libSystem.Security.Cryptography.Native.Apple.a(pal_x509.c.o)
"_SecCertificateCopySubjectSummary", referenced from:
_AppleCryptoNative_X509GetSubjectSummary in libSystem.Security.Cryptography.Native.Apple.a(pal_x509.c.o)
"_SecCertificateCreateWithData", referenced from:
_AppleCryptoNative_X509GetContentType in libSystem.Security.Cryptography.Native.Apple.a(pal_x509.c.o)
_AppleCryptoNative_X509ImportCertificate in libSystem.Security.Cryptography.Native.Apple.a(pal_x509_ios.c.o)
"_SecCertificateGetTypeID", referenced from:
_AppleCryptoNative_X509DemuxAndRetainHandle in libSystem.Security.Cryptography.Native.Apple.a(pal_x509.c.o)
"_SecCopyErrorMessageString", referenced from:
_AppleCryptoNative_SecCopyErrorMessageString in libSystem.Security.Cryptography.Native.Apple.a(pal_sec.c.o)
(maybe you meant: _p_80_plt_System_Net_Security_Interop_AppleCrypto__AppleCryptoNative_SecCopyErrorMessageString
g____PInvoke_0_0_int_llvm, _System_Security_Cryptography_Interop_AppleCrypto_AppleCryptoNative_SecCopyErrorMessageStr
ing_int , _System_Net_Security_Interop_AppleCrypto_AppleCryptoNative_SecCopyErrorMessageString_int , _p_82_plt_System
_Net_Security_Interop_AppleCrypto_AppleCryptoNative_SecCopyErrorMessageString_int_llvm , _AppleCryptoNative_SecCopyEr
rorMessageString , _p_98_plt_System_Security_Cryptography_Interop_AppleCrypto__AppleCryptoNative_SecCopyErrorMessageS
tringg____PInvoke_56_0_int_llvm , _p_99_plt_System_Security_Cryptography_Interop_AppleCrypto_AppleCryptoNative_SecCop
yErrorMessageString_int_llvm )
"_SecIdentityCopyCertificate", referenced from:
_AppleCryptoNative_X509CopyCertFromIdentity in libSystem.Security.Cryptography.Native.Apple.a(pal_x509.c.o)
"_SecIdentityCopyPrivateKey", referenced from:
_AppleCryptoNative_X509CopyPrivateKeyFromIdentity in libSystem.Security.Cryptography.Native.Apple.a(pal_x509.c.
o)
"_SecIdentityGetTypeID", referenced from:
_AppleCryptoNative_X509DemuxAndRetainHandle in libSystem.Security.Cryptography.Native.Apple.a(pal_x509.c.o)
"_SecItemCopyMatching", referenced from:
_EnumerateKeychain in libSystem.Security.Cryptography.Native.Apple.a(pal_keychain_ios.c.o)
"_SecKeyCopyAttributes", referenced from:
_AppleCryptoNative_EccGetKeySizeInBits in libSystem.Security.Cryptography.Native.Apple.a(pal_ecc.c.o)
"_SecKeyCopyExternalRepresentation", referenced from:
_AppleCryptoNative_SecKeyCopyExternalRepresentation in libSystem.Security.Cryptography.Native.Apple.a(pal_secke
y.c.o)
(maybe you meant: _AppleCryptoNative_SecKeyCopyExternalRepresentation, _p_105_plt_System_Security_Cryptography_I
nterop_AppleCrypto__AppleCryptoNative_SecKeyCopyExternalRepresentationg____PInvoke_70_0_intptr_intptr__intptr__llvm ,
_p_1624_plt_System_Security_Cryptography_Interop_AppleCrypto_AppleCryptoNative_SecKeyCopyExternalRepresentation_Syst
em_Security_Cryptography_Apple_SafeSecKeyRefHandle_Microsoft_Win32_SafeHandles_SafeCFDataHandle__Microsoft_Win32_Safe
Handles_SafeCFErrorHandle__llvm , _System_Security_Cryptography_Interop_AppleCrypto_AppleCryptoNative_SecKeyCopyExter
nalRepresentation_System_Security_Cryptography_Apple_SafeSecKeyRefHandle_Microsoft_Win32_SafeHandles_SafeCFDataHandle
_Microsoft_Win32_SafeHandles_SafeCFErrorHandle )
"_SecKeyCopyPublicKey", referenced from:
_AppleCryptoNative_EccGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_ecc.c.o)
_AppleCryptoNative_RsaGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_SecKeyCopyPublicKey in libSystem.Security.Cryptography.Native.Apple.a(pal_seckey.c.o)
(maybe you meant: _AppleCryptoNative_SecKeyCopyPublicKey)
"_SecKeyCreateDecryptedData", referenced from:
_AppleCryptoNative_RsaDecryptOaep in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_RsaDecryptPkcs in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_SecKeyCreateEncryptedData", referenced from:
_AppleCryptoNative_RsaEncryptOaep in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_RsaEncryptPkcs in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_RsaEncryptionPrimitive in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_RsaVerificationPrimitive in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_SecKeyCreateRandomKey", referenced from:
_AppleCryptoNative_EccGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_ecc.c.o)
_AppleCryptoNative_RsaGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_SecKeyCreateSignature", referenced from:
_AppleCryptoNative_RsaSignaturePrimitive in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_SecKeyCreateSignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
(maybe you meant: p_110_plt_System_Security_Cryptography_Interop_AppleCrypto__AppleCryptoNative_SecKeyCreateSig
natureg____PInvoke_76_0_intptr_byte__int_Interop_AppleCrypto_PAL_HashAlgorithm_Interop_AppleCrypto_PAL_SignatureAlgor
ithm_intptr__intptr__llvm, AppleCryptoNative_SecKeyCreateSignature , System_Security_Cryptography_Interop_AppleCryp
to_AppleCryptoNative_SecKeyCreateSignature_System_Security_Cryptography_Apple_SafeSecKeyRefHandle_System_ReadOnlySpan
1_byte_Interop_AppleCrypto_PAL_HashAlgorithm_Interop_AppleCrypto_PAL_SignatureAlgorithm_Microsoft_Win32_SafeHandles
SafeCFDataHandle__Microsoft_Win32_SafeHandles_SafeCFErrorHandle , p_112_plt_System_Security_Cryptography_Interop_Ap
pleCrypto_AppleCryptoNative_SecKeyCreateSignature_System_Security_Cryptography_Apple_SafeSecKeyRefHandle_System_ReadO
nlySpan_1_byte_Interop_AppleCrypto_PAL_HashAlgorithm_Interop_AppleCrypto_PAL_SignatureAlgorithm_Microsoft_Win32_SafeH
andles_SafeCFDataHandle__Microsoft_Win32_SafeHandles_SafeCFErrorHandle__llvm , System_Security_Cryptography_Interop
AppleCrypto_AppleCryptoNative_SecKeyCreateSignature_System_Security_Cryptography_Apple_SafeSecKeyRefHandle_byte__int
Interop_AppleCrypto_PAL_HashAlgorithm_Interop_AppleCrypto_PAL_SignatureAlgorithm_Microsoft_Win32_SafeHandles_SafeCFDa
taHandle__Microsoft_Win32_SafeHandles_SafeCFErrorHandle , p_109_plt_System_Security_Cryptography_Interop_AppleCrypt
o_AppleCryptoNative_SecKeyCreateSignature_System_Security_Cryptography_Apple_SafeSecKeyRefHandle_byte__int_Interop_Ap
pleCrypto_PAL_HashAlgorithm_Interop_AppleCrypto_PAL_SignatureAlgorithm_Microsoft_Win32_SafeHandles_SafeCFDataHandle
Microsoft_Win32_SafeHandles_SafeCFErrorHandle__llvm )
"_SecKeyCreateWithData", referenced from:
_AppleCryptoNative_SecKeyCreateWithData in libSystem.Security.Cryptography.Native.Apple.a(pal_seckey.c.o)
(maybe you meant: _p_104_plt_System_Security_Cryptography_Interop_AppleCrypto__AppleCryptoNative_SecKeyCreateWit
hDatag____PInvoke_69_0_byte__int_Interop_AppleCrypto_PAL_KeyAlgorithm_int_intptr__intptr__llvm, System_Security_Cryp
tography_Interop_AppleCrypto_AppleCryptoNative_SecKeyCreateWithData_byte__int_Interop_AppleCrypto_PAL_KeyAlgorithm_in
t_System_Security_Cryptography_Apple_SafeSecKeyRefHandle__Microsoft_Win32_SafeHandles_SafeCFErrorHandle , _AppleCryp
toNative_SecKeyCreateWithData , _p_103_plt_System_Security_Cryptography_Interop_AppleCrypto_AppleCryptoNative_SecKeyC
reateWithData_byte__int_Interop_AppleCrypto_PAL_KeyAlgorithm_int_System_Security_Cryptography_Apple_SafeSecKeyRefHand
le__Microsoft_Win32_SafeHandles_SafeCFErrorHandle__llvm )
"_SecKeyGetBlockSize", referenced from:
_AppleCryptoNative_SecKeyGetSimpleKeySizeInBytes in libSystem.Security.Cryptography.Native.Apple.a(pal_seckey.c
.o)
"_SecKeyVerifySignature", referenced from:
_AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
(maybe you meant: _AppleCryptoNative_SecKeyVerifySignature, System_Security_Cryptography_Interop_AppleCrypto_Ap
pleCryptoNative_SecKeyVerifySignature_System_Security_Cryptography_Apple_SafeSecKeyRefHandle_System_ReadOnlySpan_1_by
te_System_ReadOnlySpan_1_byte_Interop_AppleCrypto_PAL_HashAlgorithm_Interop_AppleCrypto_PAL_SignatureAlgorithm_Micros
oft_Win32_SafeHandles_SafeCFErrorHandle , System_Security_Cryptography_Interop_AppleCrypto_AppleCryptoNative_SecKey
VerifySignature_System_Security_Cryptography_Apple_SafeSecKeyRefHandle_byte__int_byte__int_Interop_AppleCrypto_PAL_Ha
shAlgorithm_Interop_AppleCrypto_PAL_SignatureAlgorithm_Microsoft_Win32_SafeHandles_SafeCFErrorHandle , p_108_plt_Sy
stem_Security_Cryptography_Interop_AppleCrypto__AppleCryptoNative_SecKeyVerifySignatureg____PInvoke_74_0_intptr_byte
_int_byte__int_Interop_AppleCrypto_PAL_HashAlgorithm_Interop_AppleCrypto_PAL_SignatureAlgorithm_intptr__llvm , _p_111
_plt_System_Security_Cryptography_Interop_AppleCrypto_AppleCryptoNative_SecKeyVerifySignature_System_Security_Cryptog
raphy_Apple_SafeSecKeyRefHandle_System_ReadOnlySpan_1_byte_System_ReadOnlySpan_1_byte_Interop_AppleCrypto_PAL_HashAlg
orithm_Interop_AppleCrypto_PAL_SignatureAlgorithm_Microsoft_Win32_SafeHandles_SafeCFErrorHandle__llvm , _p_107_plt_Sy
stem_Security_Cryptography_Interop_AppleCrypto_AppleCryptoNative_SecKeyVerifySignature_System_Security_Cryptography_A
pple_SafeSecKeyRefHandle_byte__int_byte__int_Interop_AppleCrypto_PAL_HashAlgorithm_Interop_AppleCrypto_PAL_SignatureA
lgorithm_Microsoft_Win32_SafeHandles_SafeCFErrorHandle__llvm )
"_SecPKCS12Import", referenced from:
_AppleCryptoNative_X509ImportCertificate in libSystem.Security.Cryptography.Native.Apple.a(pal_x509_ios.c.o)
"_SecPolicyCreateBasicX509", referenced from:
_AppleCryptoNative_X509ChainCreateDefaultPolicy in libSystem.Security.Cryptography.Native.Apple.a(pal_x509chain
.c.o)
"_SecPolicyCreateRevocation", referenced from:
_AppleCryptoNative_X509ChainCreateRevocationPolicy in libSystem.Security.Cryptography.Native.Apple.a(pal_x509ch
ain.c.o)
"_SecPolicyCreateSSL", referenced from:
_AppleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SecTrustCopyResult", referenced from:
_AppleCryptoNative_X509ChainGetTrustResults in libSystem.Security.Cryptography.Native.Apple.a(pal_x509chain.c.o
)
_AppleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SecTrustCreateWithCertificates", referenced from:
_AppleCryptoNative_X509ChainCreate in libSystem.Security.Cryptography.Native.Apple.a(pal_x509chain.c.o)
_AppleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SecTrustEvaluate", referenced from:
_AppleCryptoNative_X509ChainEvaluate in libSystem.Security.Cryptography.Native.Apple.a(pal_x509chain.c.o)
_AppleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SecTrustGetCertificateAtIndex", referenced from:
_AppleCryptoNative_X509ChainGetCertificateAtIndex in libSystem.Security.Cryptography.Native.Apple.a(pal_x509cha
in.c.o)
_AppleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SecTrustGetCertificateCount", referenced from:
_AppleCryptoNative_X509ChainGetChainSize in libSystem.Security.Cryptography.Native.Apple.a(pal_x509chain.c.o)
_AppleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SecTrustSetAnchorCertificates", referenced from:
_AppleCryptoNative_X509ChainSetTrustAnchorCertificates in libSystem.Security.Cryptography.Native.Apple.a(pal_x5
09chain.c.o)
_AppleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_SecTrustSetNetworkFetchAllowed", referenced from:
_AppleCryptoNative_X509ChainEvaluate in libSystem.Security.Cryptography.Native.Apple.a(pal_x509chain.c.o)
"_SecTrustSetVerifyDate", referenced from:
_AppleCryptoNative_X509ChainEvaluate in libSystem.Security.Cryptography.Native.Apple.a(pal_x509chain.c.o)
_AppleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o)
"_kSecAttrKeyClass", referenced from:
_AppleCryptoNative_SecKeyCreateWithData in libSystem.Security.Cryptography.Native.Apple.a(pal_seckey.c.o)
"_kSecAttrKeyClassPrivate", referenced from:
_AppleCryptoNative_SecKeyCreateWithData in libSystem.Security.Cryptography.Native.Apple.a(pal_seckey.c.o)
"_kSecAttrKeyClassPublic", referenced from:
_AppleCryptoNative_SecKeyCreateWithData in libSystem.Security.Cryptography.Native.Apple.a(pal_seckey.c.o)
"_kSecAttrKeySizeInBits", referenced from:
_AppleCryptoNative_EccGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_ecc.c.o)
_AppleCryptoNative_EccGetKeySizeInBits in libSystem.Security.Cryptography.Native.Apple.a(pal_ecc.c.o)
_AppleCryptoNative_RsaGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_kSecAttrKeyType", referenced from:
_AppleCryptoNative_EccGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_ecc.c.o)
_AppleCryptoNative_RsaGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_SecKeyCreateWithData in libSystem.Security.Cryptography.Native.Apple.a(pal_seckey.c.o)
"_kSecAttrKeyTypeEC", referenced from:
_AppleCryptoNative_EccGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_ecc.c.o)
"_kSecAttrKeyTypeECSECPrimeRandom", referenced from:
_AppleCryptoNative_SecKeyCreateWithData in libSystem.Security.Cryptography.Native.Apple.a(pal_seckey.c.o)
"_kSecAttrKeyTypeRSA", referenced from:
_AppleCryptoNative_RsaGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_SecKeyCreateWithData in libSystem.Security.Cryptography.Native.Apple.a(pal_seckey.c.o)
"_kSecClass", referenced from:
_EnumerateKeychain in libSystem.Security.Cryptography.Native.Apple.a(pal_keychain_ios.c.o)
"_kSecClassCertificate", referenced from:
_AppleCryptoNative_SecKeychainEnumerateCerts in libSystem.Security.Cryptography.Native.Apple.a(pal_keychain_ios
.c.o)
"_kSecClassIdentity", referenced from:
_AppleCryptoNative_SecKeychainEnumerateIdentities in libSystem.Security.Cryptography.Native.Apple.a(pal_keychai
n_ios.c.o)
"_kSecImportExportPassphrase", referenced from:
_AppleCryptoNative_X509ImportCertificate in libSystem.Security.Cryptography.Native.Apple.a(pal_x509_ios.c.o)
"_kSecImportItemIdentity", referenced from:
_AppleCryptoNative_X509ImportCertificate in libSystem.Security.Cryptography.Native.Apple.a(pal_x509_ios.c.o)
"_kSecKeyAlgorithmECDSASignatureDigestX962", referenced from:
_AppleCryptoNative_SecKeyCreateSignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
_AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
"_kSecKeyAlgorithmRSAEncryptionOAEPSHA1", referenced from:
l_switch.table.AppleCryptoNative_RsaEncryptOaep in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_kSecKeyAlgorithmRSAEncryptionOAEPSHA256", referenced from:
l_switch.table.AppleCryptoNative_RsaEncryptOaep in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_kSecKeyAlgorithmRSAEncryptionOAEPSHA384", referenced from:
l_switch.table.AppleCryptoNative_RsaEncryptOaep in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_kSecKeyAlgorithmRSAEncryptionOAEPSHA512", referenced from:
l_switch.table.AppleCryptoNative_RsaEncryptOaep in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_kSecKeyAlgorithmRSAEncryptionPKCS1", referenced from:
_AppleCryptoNative_RsaDecryptPkcs in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_RsaEncryptPkcs in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_kSecKeyAlgorithmRSAEncryptionRaw", referenced from:
_AppleCryptoNative_RsaEncryptionPrimitive in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_RsaVerificationPrimitive in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
"_kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA1", referenced from:
_AppleCryptoNative_SecKeyCreateSignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
_AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
"_kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA256", referenced from:
_AppleCryptoNative_SecKeyCreateSignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
_AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
"_kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA384", referenced from:
_AppleCryptoNative_SecKeyCreateSignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
_AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
"_kSecKeyAlgorithmRSASignatureDigestPKCS1v15SHA512", referenced from:
_AppleCryptoNative_SecKeyCreateSignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
_AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
"_kSecKeyAlgorithmRSASignatureDigestPSSSHA1", referenced from:
l_switch.table.AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_si
gnverify.c.o)
"_kSecKeyAlgorithmRSASignatureDigestPSSSHA256", referenced from:
l_switch.table.AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_si
gnverify.c.o)
"_kSecKeyAlgorithmRSASignatureDigestPSSSHA384", referenced from:
l_switch.table.AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_si
gnverify.c.o)
"_kSecKeyAlgorithmRSASignatureDigestPSSSHA512", referenced from:
l_switch.table.AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_si
gnverify.c.o)
"_kSecKeyAlgorithmRSASignatureRaw", referenced from:
_AppleCryptoNative_RsaSignaturePrimitive in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
_AppleCryptoNative_SecKeyCreateSignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
_AppleCryptoNative_SecKeyVerifySignature in libSystem.Security.Cryptography.Native.Apple.a(pal_signverify.c.o)
"_kSecMatchLimit", referenced from:
_EnumerateKeychain in libSystem.Security.Cryptography.Native.Apple.a(pal_keychain_ios.c.o)
"_kSecMatchLimitAll", referenced from:
_EnumerateKeychain in libSystem.Security.Cryptography.Native.Apple.a(pal_keychain_ios.c.o)
"_kSecReturnRef", referenced from:
_EnumerateKeychain in libSystem.Security.Cryptography.Native.Apple.a(pal_keychain_ios.c.o)
"_kSecUseDataProtectionKeychain", referenced from:
_AppleCryptoNative_EccGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_ecc.c.o)
_AppleCryptoNative_RsaGenerateKey in libSystem.Security.Cryptography.Native.Apple.a(pal_rsa.c.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : clang++ s
aiu com código 1: [C:\Sistemas\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : Undefined
symbols for architecture arm64: [C:\Sistemas\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : "_SSLCo
pyALPNProtocols", referenced from: [C:\Sistemas\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : _Ap
pleCryptoNative_SslGetAlpnSelected in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o) [C:\Sistemas\MPRAppCo
mercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : "_SSLCo
pyDistinguishedNames", referenced from: [C:\Sistemas\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : _Ap
pleCryptoNative_SslCopyCADistinguishedNames in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o) [C:\Sistemas
\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : "_SSLCo
pyPeerTrust", referenced from: [C:\Sistemas\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : _Ap
pleCryptoNative_SslCopyCertChain in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o) [C:\Sistemas\MPRAppCome
rcial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : _Ap
pleCryptoNative_SslIsHostnameMatch in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o) [C:\Sistemas\MPRAppCo
mercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : "_SSLCr
eateContext", referenced from: [C:\Sistemas\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : _Ap
pleCryptoNative_SslCreateContext in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o) [C:\Sistemas\MPRAppCome
rcial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : "_SSLGe
tNegotiatedCipher", referenced from: [C:\Sistemas\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : _Ap
pleCryptoNative_SslGetCipherSuite in libSystem.Security.Cryptography.Native.Apple.a(pal_ssl.c.o) [C:\Sistemas\MPRAppCom
ercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : "_SSLGe
tNegotiatedProtocolVersion", referenced from: [C:\Sistemas\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramework=net8.0
-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8043\targets\Xamarin.Shared.Sdk.targets(1559,3): error : _Ap
pleCryptoNative_SslGetProtocolVersion in libSystem.Se [C:\Sistemas\MPRAppComercial\MPRApp\MPRApp.csproj::TargetFramewor
k=net8.0-ios]

### Steps to Reproduce

_No response_

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 17.2

### Did you find any workaround?

If you set the linker to "no linker", the version is generated successfully. However, when sending it to the Apple Store, it is rejected

### Relevant log output

_No response_

--- a/dotnet/targets/Xamarin.Shared.Sdk.targets
+++ b/dotnet/targets/Xamarin.Shared.Sdk.targets
@@ -1492,6 +1492,9 @@
 			<!-- CFNetwork is required by xamarin_start_wwan -->
 			<_NativeExecutableFrameworks Include="CFNetwork" Condition="'$(_PlatformName)' == 'iOS'" />
 
+			<!-- libSystem.Security.Cryptography.Native.Apple.a requires the Security framework -->
+			<_NativeExecutableFrameworks Include="Security" />
+
 			<!-- Mono requires zlib, iconv, and the "Compression framework" -->
 			<_MainLinkerFlags Include="-lz" />
 			<_MainLinkerFlags Include="-liconv" />

